docs/EUDB-compliance.md

```diff
@@ -0,0 +1,65 @@
+# EUDB Guidance for 1DS C++ SDK
+
+In order to satisfy the Microsoft commitment to ensuring Privacy and Compliance, specifically
+EUDB compliance with respect to EU 'Schrems II' decision, it is imperative for certain
+commercial products to perform EUDB URL upload determination during application launch.
+
+1DS Collector service accepts data from all One Observability client SDKs. By default, traffic
+simply flows to whichever region can best handle the traffic. This approach works well for
+system required metadata. However some client scenarios require that data is sent to a specific
+geographic location only.
+
+1DS C++ SDK supports ability to specify / adjust the upload URL at runtime.
```

Comment:
Should it be at runtime, or only once during startup from `""` to (say) `url1`? I do suspect that changing at runtime from (say) `url1` to `url2` can result in events leaking to incorrect endpoints.

Reply:
Currently there's no protection that would copy the URL on start. The uploader will always "reread" the config today. Expectation here is that the product owners don't try to shoot themselves in the foot. "At runtime" here - means that we initialize, all logging functionality works, and URL is set at some point shortly after (hundreds of milliseconds to a few seconds)... What would happen if URL cannot be determined, i.e. app is offfline, or the corresponding remote service is unreachable? That is left to the product owners to decide. Currently if they set the URL to none, and it is kept that way, that's, perhaps, the most compliant action: do not attempt to upload any telemetry until you reliably know what regional collector you need to upload it to.

Whether or not someone is going to change the URL repeatedly at runtime more than once - that'd be morally wrong to do so. Example illustrates that the determination has to be performed once. Users belong to org, org belongs to fixed region. There is no scenario where user would roam between different environments or regions in my product. In case if users roam between environments (but not for EUDB), that's the option where multiple LogManagers , `LogManagerX` and `LogManagerY` can be used simultaneously.

Reply:
Current code comment before my change - says `Since this method is called for every event, you can change the URI dynamically.`, which I think is semantically similar to `ability to specify / adjust the upload URL at runtime`.

https://github.com/microsoft/cpp_client_telemetry/blob/c7bb501a159341c885c294f7d3a0301f6e8768ba/lib/api/IRuntimeConfig.hpp#L27

```
         /// <summary>
        /// Gets the URI of the collector (where telemetry events are sent).
        /// </summary>
        /// <remarks>
        /// <b>Note:</b> Since this method is called for every event, 
        /// you can change the URI dynamically.
        /// </remarks>
        /// <returns>A string that contains the collector URI.</returns>
```

Reply:
Thanks for clarifying. So to summarise " While 1ds sdk supports changing endpoint URL at runtime, the application users should be aware of its consequences.